packages/typespec-client-generator-core/src/interfaces.ts

```diff
@@ -604,3 +606,105 @@ export enum UsageFlags {
   // Set when model is used in conjunction with an application/json content type.
   Json = 1 << 8,
 }
+
+interface SdkExampleBase {
+  kind: string;
+  name: string;
+  description: string;
+  filePath: string;
+  rawExample: any;
+}
+
+export interface SdkHttpOperationExample extends SdkExampleBase {
+  kind: "http";
+  parameters: SdkHttpParameterExample[];
+  responses: Map<number, SdkHttpResponseExample>;
+}
+
+export interface SdkHttpParameterExample {
+  parameter: SdkHttpParameter;
+  value: SdkTypeExample;
+}
+
+export interface SdkHttpResponseExample {
+  response: SdkHttpResponse;
+  headers: SdkHttpResponseHeaderExample[];
+  value?: SdkTypeExample;
+}
+
+export interface SdkHttpResponseHeaderExample {
+  header: SdkServiceResponseHeader;
+  value: SdkTypeExample;
+}
+
+export type SdkTypeExample =
+  | SdkStringExample
+  | SdkNumberExample
+  | SdkBooleanExample
+  | SdkNullExample
+  | SdkAnyExample
+  | SdkArrayExample
+  | SdkDictionaryExample
+  | SdkUnionExample
+  | SdkModelExample;
+
+export interface SdkExampleTypeBase {
+  kind: string;
+  type: SdkType;
+  value: unknown;
+}
+
+export interface SdkStringExample extends SdkExampleTypeBase {
+  kind: "string";
+  type: SdkBuiltInType | SdkDateTimeType | SdkDurationType | SdkEnumType | SdkConstantType;
+  value: string;
+}
+
+export interface SdkNumberExample extends SdkExampleTypeBase {
+  kind: "number";
+  type: SdkBuiltInType | SdkDateTimeType | SdkDurationType | SdkEnumType | SdkConstantType;
+  value: number;
+}
+
+export interface SdkBooleanExample extends SdkExampleTypeBase {
+  kind: "boolean";
+  type: SdkBuiltInType | SdkConstantType;
+  value: boolean;
+}
+
+export interface SdkNullExample extends SdkExampleTypeBase {
+  kind: "null";
+  type: SdkNullableType;
+  value: null;
+}
+
+export interface SdkAnyExample extends SdkExampleTypeBase {
+  kind: "any";
+  type: SdkBuiltInType;
+  value: unknown;
+}
+
+export interface SdkArrayExample extends SdkExampleTypeBase {
+  kind: "array";
+  type: SdkArrayType;
+  value: SdkTypeExample[];
+}
+
+export interface SdkDictionaryExample extends SdkExampleTypeBase {
+  kind: "dict";
+  type: SdkDictionaryType;
+  value: Record<string, SdkTypeExample>;
+}
+
+export interface SdkUnionExample extends SdkExampleTypeBase {
+  kind: "union";
+  type: SdkUnionType;
+  value: unknown;
+}
+
+export interface SdkModelExample extends SdkExampleTypeBase {
+  kind: "model";
+  type: SdkModelType;
+  value: Record<string, SdkTypeExample>;
+  additionalProperties?: Record<string, SdkTypeExample>;
```

Comment:
why additionalProperties

Reply:
it represent any additional value for model with additional properties.